modules/planning/common/path/discretized_path.cc

```diff
@@ -52,10 +52,10 @@ PathPoint DiscretizedPath::Evaluate(const double path_s) const {
                                                            *it_lower, path_s);
 }
 
-std::vector<PathPoint>::const_iterator DiscretizedPath::QueryLowerBound(
-    const double path_s) const {
+std::vector<PathPoint>::const_iterator
+DiscretizedPath::QueryLowerBoundbyAbsoluteS(const double path_s) const {
   auto func = [](const PathPoint &tp, const double path_s) {
-    return tp.s() < path_s;
+    return std::fabs(tp.s()) < std::fabs(path_s);
```

Comment:
This will not work when both positive and negative s exist. std::lower_bound requires a sorted container using the comparator.